pkg/trace/stats/aggregation_test.go

```diff
@@ -57,33 +57,57 @@ func TestNewAggregation(t *testing.T) {
 	for _, tt := range []struct {
 		in               *pb.Span
 		enablePeerSvcAgg bool
-		res              Aggregation
+		resAgg           Aggregation
+		resPeerTags      []string
 	}{
 		{
 			&pb.Span{},
 			false,
 			Aggregation{},
+			nil,
 		},
 		{
 			&pb.Span{},
 			true,
 			Aggregation{},
+			nil,
 		},
 		{
 			&pb.Span{
 				Service: "a",
-				Meta:    map[string]string{"peer.service": "remote-service"},
+				Meta:    map[string]string{"span.kind": "client", "peer.service": "remote-service"},
 			},
 			false,
+			Aggregation{BucketsAggregationKey: BucketsAggregationKey{Service: "a", SpanKind: "client"}},
+			nil,
+		},
+		// peer.service stats aggregation is enabled, but span.kind != (client, producer).
+		{
+			&pb.Span{
+				Service: "a",
+				Meta:    map[string]string{"span.kind": "", "peer.service": "remote-service"},
```

Comment:
I suggest one where `span.kind` is empty (as you have) but another where `span.kind` is non-empty but set to something other than `client` or `producer`